tests/factories/cht/generate.js

```diff
@@ -51,7 +57,71 @@ const createData = ({ healthCenter, user, nbrClinics=10, nbrPersons=10 }) => {
   return { clinics, reports, persons };
 };
 
-const createHierarchy = ({ name, user=false, nbrClinics=50, nbrPersons=10 }) => {
+const createDataWithRealNames = ({ healthCenter, user, nbrClinics = 10, nbrPersons = 10 }) => {
+  const clinics = Array.from({ length: nbrClinics }).map((_, index) => {
+    const firstName = firstNames[index % firstNames.length];
+    const lastName = lastNames[index % lastNames.length];
+    const personName = `${firstName} ${lastName}`;
+    const personPhoneNumber = phoneNumbers[index % phoneNumbers.length];
+
+    // Create primary contact
+    const contact = {
+      _id: 'fixture:primary:person',
+      name: personName,
+      phone: personPhoneNumber
+    };
+
+    // Create the clinic with the name based on the primary person's name
+    const clinic = placeFactory.place().build({
+      type: 'clinic',
+      parent: { _id: healthCenter._id, parent: healthCenter.parent },
+      name: `${personName} Family`,
+      contact: contact
+    });
+
+    // Create the primary contact for this clinic
+    const primaryPerson = personFactory.build({
+      parent: { _id: clinic._id, parent: clinic.parent },
+      name: personName,
+      phone: personPhoneNumber
+    });
+
+    return { clinic, primaryPerson };
+  });
+
+  const persons = clinics.flatMap(({ clinic }) => Array.from({ length: nbrPersons - 1 }).map((_, index) => {
+    const additionalPersonName = `${firstNames[index % firstNames.length]} ${lastNames[index % lastNames.length]}`;
+    const additionalPhoneNumber = phoneNumbers[index % phoneNumbers.length];
+    return personFactory.build({
+      parent: { _id: clinic._id, parent: clinic.parent },
+      name: additionalPersonName,
+      patient_id: 65421,
+      phone: additionalPhoneNumber
+    });
+  }));
+
+  const allPersons = [
+    ...clinics.map(({ primaryPerson }) => primaryPerson),
+    ...persons
+  ];
+
+  const reports = allPersons.flatMap(person => [
```

Comment:
At this point, I think allPersons is a flat - regular array, so where we can do 
```suggestion
  const reports = [];
  allPersons.forEach(person => reports.add(...[
```
Just trying to make it more readable, but see if this suggestion really helps :)